content/en/samples/posts/2024-12-26-idle-view-app.md

```diff
@@ -0,0 +1,444 @@
+| `react-dom`           | ^18.3.1         | Matches React version ^18.x  | Required for React rendering. |
+| `vite`                | ^5.4.9          | Node.js >=16.0               | Bundler; needs Node.js v16+.  |
+| `tailwindcss`         | ^3.4.14         | Works with PostCSS ^8.4      | Styling framework.            |
+| `vite-plugin-svgr`    | ^4.3.0          | Compatible with Vite ^5.x    | For SVG imports as components.|
+
+### Backend Dependencies
+| Dependency Name       | Current Version | Required/Peer Version(s)        | Notes                          |
+|------------------------|-----------------|---------------------------------|--------------------------------|
+| `prisma`              | ^5.21.1         | Node.js >=16.x, `@prisma/client` must match | ORM for database.             |
+| `express`             | ^4.21.1         | Works with Node.js >=12.x      | Backend framework.            |
+| `jsonwebtoken`        | ^9.0.2          | Requires `bcryptjs` for hashing| For authentication.           |
+| `mysql2`              | ^3.11.3         | Works with MySQL >=8.x         | Database connector.           |
+
+
+## Project Setup
+This section provides a step-by-step guide to setting up the IdleView project, covering both backend and frontend configurations.
+
+### General Setup
+1. **Clone the repository.**
+   ```sh
+   git clone https://github.com/seangolden11/Idle-View-team7.git
+   cd Idle-View-team7
+   ```
+
+#### Repository Structure
+
+The repository is organized into several directories, each serving a specific purpose:
+
+- **backend**: Contains the server-side code and related resources for the project.
+- **frontend**: Houses the client-side code, including the user interface components and associated assets.
+- **회의록**: (Korean for "meeting records") Contains records of team meetings and discussions.
+- **Project Related Files**: Include miscellaneous files or documents relevant to the project.
+
+Additionally, there are files in the root directory:
+- **.gitattributes**: Defines attributes for pathnames to customize repository behaviors, such as end-of-line handling.
+- **.gitignore**: Specifies files and directories to be ignored by Git, preventing them from being tracked.
+- **README.md**: Provides an overview of the project, including setup instructions and usage guidelines.
+
+---
+
+### Backend Setup
+
+1. **Navigate to the backend directory:**
+   ```sh
+   cd backend
+   ```
+2. **Install dependencies:**
+   ```sh
+   npm install
+   ```
+3. **Create `.env` file:**
+   Create an `.env` file in the backend directory and configure the environment variables. Example `.env` file content:
+   ```plaintext
+   DATABASE_URL="mysql://root:password@localhost:3306/idleview"
+   JWT_SECRET=awjdlksejnlsdjgslgang/4ksjfskdvn= # Replace with any random secure string
+   WEATHER_API_KEY=4c4c552d80ea31da2d4e01e48bc04a61 # Obtain your API key from a weather API provider
```

Comment:
Are the three values fixed? If they need to be set individually, please provide an explanation for each one.